linkerd/tls/src/server.rs

```diff
@@ -17,19 +19,39 @@ use tokio_rustls::server::TlsStream;
 use tower::util::ServiceExt;
 use tracing::{debug, trace, warn};
 
-pub trait HasConfig {
-    fn tls_server_name(&self) -> identity::Name;
-    fn tls_server_config(&self) -> Arc<Config>;
-}
+pub type Config = Arc<rustls::ServerConfig>;
 
 /// Produces a server config that fails to handshake all connections.
-pub fn empty_config() -> Arc<Config> {
+pub fn empty_config() -> Config {
     let verifier = rustls::NoClientAuth::new();
-    Arc::new(Config::new(verifier))
+    Arc::new(rustls::ServerConfig::new(verifier))
+}
+
+/// A marker type for remote client idenities..
```

Comment:
nit: ...and again, I probably wouldn't call this a "marker", since it _does_ contain actual data?